src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -2297,6 +2297,18 @@ private BoundExpression BindNameofOperatorInternal(InvocationExpressionSyntax no
             {
                 diagnostics.AddAssembliesUsedByNamespaceReference(nsExpr.NamespaceSymbol);
             }
+            else if (boundArgument is BoundIndexerAccess indexerAccess)
+            {
+                // nameof(x[y]) where the argument is an indexer is an error. Ref analysis requires
+                // the indexer to be recognized as get, set, or both. We don't know which was
+                // intended for nameof() but since this is already an error, assume it's get.
+                bool setAccessorKind = !syntaxIsOk && indexerAccess.AccessorKind == AccessorKind.Unknown;
+                Debug.Assert(setAccessorKind);
```

Comment:
If this assert fails does the compiler still throw as it does today?

Reply:
If the assert fails with `syntaxIsOk == true` and `indexerAccess.AccessorKind == AccessorKind.Unknown`, then I expect the compiler will still throw an exception.

I could change the `Debug.Assert` and corresponding `if` statement to only check `indexerAccess.AccessorKind` to prevent the exception if that seems more appropriate.